### PR TITLE
Track available accounts

### LINKED ISSUE #18: Track available accounts
Add the ability to track available AWS accounts.  When an aws-account is requested, check the available pool first instead of creating a new account.  When the service is deleted, return the account to the to the available pool.

### PR DESCRIPTION
**Pull request by briankfitzwater**
- Marks account as "unassigned" when the broker manager requests the service be deprovisioned
- Looks for accounts marked "available" and assigns those to the service instead of creating account
- Closes #18

### CODE REVIEW

File: accounts.go [hunk 1]
```diff
@@ -58,6 +58,22 @@ func (am accountManager) CreateAccount(acctName string, email string, db *gorm.D
 		return nil, err
 	}
 
+	var instance serviceInstance
+	err = db.First(&instance, "instance_id = ?", "available").Error
```
Comment (afeld) [inline accounts.go:62]: Maybe we should use `null` rather than a special string?

File: main.go [hunk 2]
```diff
@@ -78,8 +78,14 @@ func main() {
...
 
-	port := os.Getenv("PORT")
-	origin := ":" + port
+	var origin string
+	port, found := os.LookupEnv("PORT")
+	if found {
+		origin = ":" + port
+	} else {
+		origin = "localhost:8080"
+	}
```
Comment (afeld) [inline main.go:87]: Nice.

File: accounts.go [hunk 3]
```diff
@@ -58,6 +58,22 @@ func (am accountManager) CreateAccount(acctName string, email string, db *gorm.D
...
+
+	if err == nil {
```
Comment (afeld) [inline accounts.go:64]: More Go-y to do this the other way around: check for `err` and return it if not `nil`.

**Review by afeld — COMMENTED** [review]
The code looks pretty good, but I thought we decided not to do this...? https://gsa-tts.slack.com/archives/C75VCSYHG/p1522930332000486?thread_ts=1522413425.000204&cid=C75VCSYHG

**Comment by briankfitzwater** [comment]
After further discussion with @aidanfeldman we've decided not to do this.  I'm cancelling the pull request.  I'm keeping the branch because I need part of this functionality in order to demo/test the broker.  Without the ability to delete services, I can't remove the service broker when I'm done testing.

**Pull request closed without merging**